src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -1012,7 +1012,15 @@ public virtual object Clone()
                 ci._textInfo = (TextInfo)_textInfo.Clone();
             }
 
-            if (_calendar != null)
+            if (_dateTimeInfo != null && _dateTimeInfo.Calendar == _calendar)
+            {
+                // Usually when we access CultureInfo.DateTimeFormat first time, we create the DateTimeFormatInfo object
+                // using CultureInfo.Calendar. i.e. CultureInfo.DateTimeInfo.Calendar == CultureInfo.calendar.
+                // When cloning CultureInfo, if we know this still the case CultureInfo.DateTimeInfo.Calendar == CultureInfo.calendar
+                // then we can keep the same behavior for the cloned object and no need to create another calendar object.
+                ci._calendar = ci.DateTimeFormat.Calendar;
```

Comment:
is calendar immutable? if it is can we also just copy ref in the branch below?

Reply:
Yes calendar is immutable. And this will be wrong because cloning guarantee to get a new immutable copy of the calendar. 

---
In reply to: [324474221](https://github.com/dotnet/coreclr/pull/26721#discussion_r324474221) [](ancestors = 324474221)

Reply:
Will this work if you clone twice? won't you get the same ref? should this be ci._calendar = ci.DateTimeFormat.Calendar.Clone()?

Reply:
No, when cloning, if DateTimeFormatInfo is not null, we clone DateTimeFormatInfo too which will clone the calendar object there too. The whole point of the fix is we need to have ci.Calnedar and ci.DateTimeFormat.Calendar points to the same object after cloning if it was pointing to same instance before cloning. 

I mean if we have:

ci.Calnedar == ci.DateTimeFormat.Calendar
ci1 = (CultureInfo) ci.Clone();

then the following should be true

ci1.Calnedar == ci1.DateTimeFormat.Calendar

Reply:
I missed the fact that we clone DateTimeFormat too - thanks for clarifying!